### PR TITLE
fix(history): Heal duplicate peer entries with different case

### PR DESCRIPTION
**Pull request by anthonybilinski**
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/6611)

### CODE REVIEW

**Comment by anthonybilinski** [comment]
@Chiitoo this should fix the issue you were seeing in the peer split upgrade.

**Comment by Chiitoo** [comment]
Very nice.  Looks good to me so far!

```
persistence/db/upgrades/dbupgrader.cpp:257 : Debug: Database upgraded incrementally to schema version  11
persistence/db/upgrades/dbupgrader.cpp:260 : Info: Database upgrade finished (databaseSchemaVersion 10 -> 11 )
```

Many thanks!

File: src/persistence/db/upgrades/dbto11.cpp
Comment (robinlinden) [inline src/persistence/db/upgrades/dbto11.cpp]: Should the appendBlah calls in here have their failure-bools propagated up? If not, should they still be returning them? :P

**Review by robinlinden — APPROVED** [review]
`euqal` -> `equal` in the ad6fb0b2d631ed8b090d223e0efd44cbfb1fa78e commit message.
d974631299e571a3a7618f6ece6adae46ade8bc8 and 463ab46d7f06a5c832e3f4d122d91b08a9e9c16e look like they need their commit messages wrapped. (Usually this part of the review is handled by some git commit linter. :P There are lots for conventional commits.)

I think things look fine, but since I don't really work on qTox and this touches the db, you might want to wait for another review.

**Review by anthonybilinski — COMMENTED** [review]
Fixed spelling and line wrapping in commit messages. Definitely should be handled automatically :S Tacked on to https://github.com/qTox/qTox/issues/6513.

@sudden6 if you want to give this a second look before merging go ahead, if you're happy with Robin's review I can merge without too.

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6611#-:-N1Y41kHEnsnRPawdfMj:b-iuas2q)* status: 0 of 1 LGTMs obtained

___
*[`src/persistence/db/upgrades/dbto11.cpp` line 30 at r4](https://reviewable.io/reviews/qtox/qtox/6611#-MzYCvws0LUM_wx7NvZc-r4-30:-N1Y4aRt1GEZXR939bve:b-dul6hw) ([raw file](https://github.com/qtox/qtox/blob/ad6fb0b2d631ed8b090d223e0efd44cbfb1fa78e/src/persistence/db/upgrades/dbto11.cpp#L30)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

Should the appendBlah calls in here have their failure-bools propagated up? If not, should they still be returning them? :P
</blockquote></details>

Arg again. Yes they should be handled, propagated up now.

**Review by robinlinden — COMMENTED** [review]
265398a44e56b849bbd53c1739b57a5eddbe3795 and 00119ff0d415b0499c73d761a7d6c841ebbd73eb have the exact same commit message. :x Probably not intentional?

**Review by anthonybilinski — COMMENTED** [review]
:facepalm: fixed.

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6611#-:-N1YBGFe5VxI_qOinCrD:bjephok)* status: 0 of 1 LGTMs obtained

**Review by sudden6 — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r11, 2 of 3 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6611)* status: 0 of 1 LGTMs obtained

___
*[`src/persistence/db/upgrades/dbupgrader.cpp` line 50 at r12](https://reviewable.io/reviews/qtox/qtox/6611#-N2_z6Kk3I_R9clJ4kJS:-N2_z6Kk3I_R9clJ4kJT:bds6016) ([raw file](https://github.com/qtox/qtox/blob/7a8eee7d8e25c74fbcf8f0d355fb03c92f60be90/src/persistence/db/upgrades/dbupgrader.cpp#L50)):*
> ```cpp
>     RowId validPeerRow;
>     db.execNow(RawDatabase::Query(QStringLiteral("SELECT id FROM peers WHERE CAST(public_key AS BLOB)=?;"),
>                                     {chatId.toString().toUtf8()},
> ```

Why convert `chatId` from binary, to string, back to binary?

**Review by anthonybilinski — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6611)* status: 0 of 1 LGTMs obtained

___
*[`src/persistence/db/upgrades/dbupgrader.cpp` line 50 at r12](https://reviewable.io/reviews/qtox/qtox/6611#-N2_z6Kk3I_R9clJ4kJS:-N2dC6dg5MNvRT05og6g:b-93n2fk) ([raw file](https://github.com/qtox/qtox/blob/7a8eee7d8e25c74fbcf8f0d355fb03c92f60be90/src/persistence/db/upgrades/dbupgrader.cpp#L50)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

Why convert `chatId` from binary, to string, back to binary?
</blockquote></details>

In the dbTo11 upgrade, we're trying to get the uppercase string that matches our current lowercase string if one exists, and if not insert it. If we just select with chatId.getByteArray(), we will get back the lower case string always since we're doing a binary comparison with the CAST.

Doing .toString() gives us the uppercase version of the ascii representation of the bytearray, so allows us to check specifically for that.

This is covered by the dbTo11 unit test, but I've added a comment in as well to make sure this doesn't get changed in the future.

**Review by sudden6 — COMMENTED** [review]
with the typo fixed: :lgtm_strong: 

> Reviewed 1 of 2 files at r15.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6611#-:-N2gUe0P4_czcNXr58Hc:b-h9d2fk)* status: :shipit: complete! 1 of 1 LGTMs obtained

___
*[`src/persistence/db/upgrades/dbupgrader.cpp` line 50 at r12](https://reviewable.io/reviews/qtox/qtox/6611#-N2_z6Kk3I_R9clJ4kJS:-N2gURGiDNoA4qngKaF3:bdd07ox) ([raw file](https://github.com/qtox/qtox/blob/7a8eee7d8e25c74fbcf8f0d355fb03c92f60be90/src/persistence/db/upgrades/dbupgrader.cpp#L50)):*
<details><summary><i>Previously, anthonybilinski (Anthony Bilinski) wrote…</i></summary><blockquote>

In the dbTo11 upgrade, we're trying to get the uppercase string that matches our current lowercase string if one exists, and if not insert it. If we just select with chatId.getByteArray(), we will get back the lower case string always since we're doing a binary comparison with the CAST.

Doing .toString() gives us the uppercase version of the ascii representation of the bytearray, so allows us to check specifically for that.

This is covered by the dbTo11 unit test, but I've added a comment in as well to make sure this doesn't get changed in the future.
</blockquote></details>

Yeah that was a bit confusing. Typo: `string`

**Comment by anthonybilinski** [comment]
@sudden6, typo fixed but merge is blocked by a requested change of yours still.

**Review by sudden6 — APPROVED** [review]

**Pull request merged**
